cfme/tests/infrastructure/test_vm_ownership.py

```diff
@@ -85,3 +90,38 @@ def test_set_vm_owner(self, appliance, vm, from_detail):
         rest_vm.reload()
         assert hasattr(rest_vm, "evm_owner_id")
         assert rest_vm.evm_owner.userid == "admin"
+
+
+@pytest.fixture(scope='module')
+def small_vm(provider, small_template_modscope):
+    vm = provider.appliance.collections.infra_vms.instantiate(random_vm_name(context='rename'),
+                                                              provider,
+                                                              small_template_modscope.name)
+    vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+    vm.refresh_relationships()
+    yield vm
+    if vm.exists:
+        vm.cleanup_on_provider()
+
+
+@pytest.mark.ignore_stream('5.9')
+@pytest.mark.provider([VMwareProvider], override=True, scope="module", selector=ONE_PER_TYPE)
+def test_rename_vm(small_vm):
+
+    """Test for rename the VM.
+       This feature is included in 5.10z.
+    Steps:
+    1. Add VMware provider
+    2. Provision VM
+    3. Navigate to details page of VM
+    4. Click on Configuration > Rename this VM > Enter new name
+    5. Click on submit
+    6. Check whether VM is renamed or not
+    """
+    view = navigate_to(small_vm, 'Details')
+    changed_vm = small_vm.rename_vm(new_vm_name="test-{}".format(fauxfactory.gen_alphanumeric()))
+    view.flash.wait_displayed("20s")
+    msg = 'Rename of Virtual Machine "{vm_name}" has been initiated'.format(vm_name=small_vm.name)
+    view.flash.assert_success_message(msg)
+    assert changed_vm.exists
+    changed_vm.rename_vm(small_vm.name)
```

Comment:
If you actually modify the `name` attribute of the InfraVm object, instead of instantiating a new one, then the cleanup will work just fine without having to do this rename back to the original.

Reply:
Thank you @mshriver 
This modification saved more time which was taken to perform renaming operation on VM again.